src/Installers/Windows/AspNetCoreModule-Setup/ANCMV2/aspnetcoremodulev2.wxs

```diff
@@ -283,7 +283,7 @@
             <AppSearch Before="LaunchConditions" />
             <LaunchConditions After="FindRelatedProducts" />
             <MigrateFeatureStates />
-            <Custom Action="CheckForSharedConfiguration" After="InstallInitialize">OPT_NO_SHARED_CONFIG_CHECK=0 AND NOT Installed AND NOT REMOVE="ALL"</Custom>
+            <Custom Action="CheckForSharedConfiguration" After="InstallInitialize">(NOT OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK="" OR OPT_NO_SHARED_CONFIG_CHECK="0") AND NOT Installed AND NOT REMOVE="ALL"</Custom>
```

Comment:
Could you explain this redundant-seeming expression please❔ I _think_ it means `OPT_NO_SHARED_CONFIG_CHECK` is not defined, empty, or `0` but it's not clear why undefined or empty can happen now. Might be worth a comment in the file too.

Does the change at https://github.com/dotnet/aspnetcore/pull/37969/files#diff-89e82b5e94dfdfcebd70f2f290c0a47cc54341c6e868577a9f4ad82a74d3ba73L60-R66 explain things❔ If yes, definitely worth a comment at least covering order of execution. Why hasn't the new `HostOptions` installer done its magic by this point❔ Probably need information in every spot where undefined or empty are now possible (because it's weird)

Reply:
Yes, that's why. I'll add comments. `HostOptions` executes separately as it's a separate .msi - all it does is write to the registry. The bundle is where we figure out the value for each option, but it can leave the options empty if there's nothing in the registry and the user passes nothing. We treat empty as 0 for these options now, hence the change.